qiskit_optimization/applications/knapsack.py

```diff
@@ -0,0 +1,69 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An application class for the Knapsack problem"""
+from typing import List, Union
+
+import numpy as np
+from docplex.mp.model import Model
+
+from qiskit_optimization.algorithms import OptimizationResult
+from qiskit_optimization.problems.quadratic_program import QuadraticProgram
+from .optimization_application import OptimizationApplication
+
+
+class Knapsack(OptimizationApplication):
+    """Optimization application for the "knapsack problem" [1].
+
+    References:
+        [1]: "Knapsack problem",
+        https://en.wikipedia.org/wiki/Knapsack_problem
+    """
+
+    def __init__(self, values: List[int], weights: List[int], max_weight: int) -> None:
+        """
+        Args:
+            values: A list of the values of items
+            weights: A list of the weights of items
+            max_weight: The maximum weight capacity
+        """
+        self._values = values
+        self._weights = weights
+        self._max_weight = max_weight
```

Comment:
I see some applications allow aspects to be changed. Like in clique `size` is changeable, and in vehicle routine `num_vehicles` and `depot`  are.  It seems `max_weight` could be settable the same way, so given the same set of items, try the same problem with different max weights, like trying vehicle routing with a different number of vehicles...